libraries/AP_Scripting/applets/Param_Controller.lua

```diff
@@ -26,51 +35,127 @@ function get_scripts_dir()
    return "scripts"
 end
 
+--[[
+   read file in chunks to avoid timeout
+--]]
+function read_file_chunk()
+   if not file_reading or not file_handle then
+      return false
+   end
+
+   local chunk_size = 200
+   for _ = 1, chunk_size do
+      local line = file_handle:read()
+      if not line then
+         file_handle:close()
+         file_handle = nil
+         file_reading = false
+         param_loading = true
+         return false
+      end
+      table.insert(param_lines, line)
+   end
+   return true
+end
+
 --[[
 load parameters from a file PARAM_FILENAME from directory n
 --]]
 function param_load(n)
-   count = 0
-   failed = false
    file_name = get_scripts_dir() .. "/" .. n .."/" .. PARAM_FILENAME
   -- Open file
-  file = io.open(file_name)
-  if not file then
+  file_handle = io.open(file_name)
+  if not file_handle then
      gcs:send_text(0,string.format("%s not present",file_name))
      return
   end
-   while true do
-      local line = file:read()
-      if not line then
-         break
+
+  gcs:send_text(6,string.format("Loading config %d",n))
+
+  -- Initialize for chunked reading
+  param_lines = {}
+  param_index = 1
+  param_loading = false
+  file_reading = true
+  param_subdir = n
+end
+
+--[[
+   process parameters in batches to avoid timeout
+--]]
+function param_load_batch()
+   if not param_loading then
+      return false
+   end
+
+   local count = 0
+   local failed = false
+   local batch_size = 200
+
+   for _ = 1, batch_size do
+      if param_index > #param_lines then
+         param_loading = false
+         if not failed then
+            gcs:send_text(6,string.format("Config %d loaded", param_subdir))
+         else
+            gcs:send_text(6,string.format("Config %d loaded (some failed)", param_subdir))
+         end
+         return false
       end
+
+      local line = param_lines[param_index]
+      param_index = param_index + 1
+
       -- trim trailing spaces
       line = string.gsub(line, '^(.-)%s*$', '%1')
-      local _, _, parm, value = string.find(line, "^([%w_]+)%s*([%d]*.[%d]*)")
-      if parm then
-         if not param:set(parm,value) then
-            failed = true
-         else
-            count = count +1
+
+      -- skip empty lines and comments
+      if line ~= "" and not string.match(line, "^%s*#") then
+         -- Try comma-separated format first (PARAM,value)
+         local _, _, parm, value = string.find(line, "^([%w_]+),([%d.-]+)")
+         if not parm then
+            -- Fall back to space-separated format (PARAM value)
+            _, _, parm, value = string.find(line, "^([%w_]+)%s+([%d.-]+)")
+         end
```

Comment:
I think you could use `"^([%w_]+)[, ] *([%d.-]+)"` to work for both? A `,` or ` ` followed by zero or more ` `?